packages/safe-kit/src/utils/sendTransaction.ts

```diff
@@ -17,15 +17,53 @@ export const sendTransaction = async ({
   transaction: Transaction
   protocolKit: Safe
 }): Promise<string | undefined> => {
-  const signer = (await protocolKit
-    .getSafeProvider()
-    .getExternalSigner()) as unknown as AbstractSigner
-  const txResponsePromise = await signer.sendTransaction({
-    from: (await protocolKit.getSafeProvider().getSignerAddress()) || '0x',
-    ...transaction
+  const signer = (await protocolKit.getSafeProvider().getExternalSigner()) as WalletClient<
+    Transport,
+    Chain,
+    Account
+  >
+  const client = await protocolKit.getSafeProvider().getExternalProvider()
+
+  if (!signer)
+    throw new Error('SafeProvider must be initialized with a signer to use this function')
+
+  const hash = await signer.sendTransaction({
+    to: transaction.to as Address,
+    data: transaction.data as Hex,
+    value: BigInt(transaction.value),
+    account: signer.account
   })
 
-  const txResponse = await txResponsePromise.wait()
+  const receipt = await client.waitForTransactionReceipt({ hash })
+
+  return receipt.transactionHash
+}
+
+export function createLegacyTxOptions(options?: TransactionOptions) {
```

Comment:
Have we got an util in the protocol-kit for this?

Reply:
Yes we do, but there was a reason why Leo didn't use it, he just didn't remember. We can check it again

Reply:
The difference seems to be only that this function here additionally defines `maxFeePerGas` and `maxPriorityFeePerGas` compared to the one in `packages/protocol-kit/src/utils/transactions/utils.ts`. Maybe this was the reason

<img width="1360" alt="Screenshot 2024-07-25 at 14 48 29" src="https://github.com/user-attachments/assets/efcecfd1-17a9-49d5-bfa4-fb8e1c218b7c">

Reply:
After checking further, I could not find any use of the `createLegacyTxOptions` function defined here. Is it really needed or maybe just a forgotten leftover after a refactor? @leonardotc

Reply:
Yeah, probably. I removed a lot of duplicates. I possibly forgot that one.